tunder/lib/src/database/query.dart

```diff
@@ -127,6 +127,12 @@ class Query<T> {
     return _where;
   }
 
+  Query<T> whereMap(Map<String, dynamic> map) {
```

Comment:
I wish Dart had [union-like APIs](https://github.com/dart-lang/language/issues/145)/union types, then the `where` method could have a `String | Map<String, dynamic>` parameter. Yet `where(String param1)` and `whereMap(Map<String, dynamic> param1)` may document the API better than `where(dynamic param1)`. Do you prefer `whereMap` (a shorter name) over `whereColumns`?

Reply:
Exactly. That's why I didn't use dynamic params for the `where` method. I also wish union types for Dart. I hope it won't take too much time to have these types available.